internal/cache/util/util_test.go

```diff
@@ -233,3 +235,50 @@ func (ut *utilTest) Test_IsCacheHandleValid_False() {
 		ExpectFalse(IsCacheHandleInvalid(errors.New(errMsg)))
 	}
 }
+
+func TestCheckPermissionsForShouldReturnTrueWhenDirectoryExists(t *testing.T) {
+	base := "./" + generateRandomString()
+	dirPath := base + "/" + "path/cachedir"
+	dirCreationErr := os.MkdirAll(dirPath, 0700)
+	defer os.RemoveAll(base)
+
+	hasPermissions, err := CheckCacheDirectoryPermissions(dirPath)
+
+	ExpectEq(nil, dirCreationErr)
+	ExpectEq(nil, err)
+	ExpectTrue(hasPermissions)
+}
+
+func TestCheckPermissionsForShouldReturnTrueWhenDirectoryCanBeCreated(t *testing.T) {
+	base := "./" + generateRandomString()
+	dirPath := base + "/" + "path/cachedir"
+	defer os.RemoveAll(base)
+
+	hasPermissions, err := CheckCacheDirectoryPermissions(dirPath)
+
+	ExpectEq(nil, err)
+	ExpectTrue(hasPermissions)
+}
+
+func TestCheckPermissionsForShouldReturnFalseWithErrorWhenDirectoryDoesNotHavePermissions(t *testing.T) {
+	dirPath := "./" + generateRandomString()
+	dirCreationErr := os.MkdirAll(dirPath, 0444)
+	defer os.RemoveAll(dirPath)
+
+	hasPermissions, err := CheckCacheDirectoryPermissions(dirPath)
+
+	ExpectEq(nil, dirCreationErr)
+	ExpectFalse(hasPermissions)
+	ExpectNe(nil, err)
+	ExpectTrue(strings.Contains(err.Error(), "error creating file at directory ("+dirPath+"), error"))
+}
+
+func generateRandomString() string {
```

Comment:
we can reuse this method https://github.com/GoogleCloudPlatform/gcsfuse/blob/read_cache_release/internal/util/test_util.go#L22

Reply:
done with latest commit